client/components/content-elements/tce-video/edit/index.vue

```diff
@@ -64,36 +63,19 @@ export default {
     isFocused: { type: Boolean, default: false },
     isDragged: { type: Boolean, default: false }
   },
-  data() {
-    return {
-      error: null,
-      switchingVideo: false
-    };
-  },
+  data: () => ({ error: null, switchingVideo: false }),
   computed: {
-    player() {
-      return get(this.$refs, 'video.player');
-    },
-    url() {
-      return get(this.element, 'data.url', '');
-    },
-    type() {
-      if (NOT_NATIVE.test(this.url)) return { isNative: false };
-      const url = this.url.split('?').shift();
-      const ext = extname(url).substring(1);
-      const name = `video/${CUSTOM_SUBTYPE_MAPPING[ext] || ext}`;
-      return { isNative: true, name };
+    player: ({ $refs }) => get($refs, 'video.player'),
+    url: ({ element }) => get(element, 'data.url', ''),
+    video: ({ url }) => {
+      url = new URL(url);
+      if (isDownloadLink(url)) return { url: url.href, native: true, mime: 'video/*' };
+      if (isShareLink(url)) return { url: url.href, native: false };
+      return { url: url.href, native: true, mime: mimetype(url) };
```

Comment:
Consider replacing with something like:
```
return {
  url: url.href,
  native: !isShareLink(url),
  mime: isX && mimetype(url)
}
```
Since some stuff is being repeated

Reply:
👍